sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -36,6 +36,12 @@ import org.apache.spark.util.random.{BernoulliCellSampler, PoissonSampler}
 case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
   extends UnaryExecNode with CodegenSupport {
 
+  @transient private lazy val aliasesInProjects = projectList.flatMap(_.collectFirst {
```

Comment:
Could we fix the issue in `EnsureRequirements`? Aggregate operators can also introduce alias.